charts/linkerd2/templates/proxy-injector.yaml

```diff
@@ -33,9 +33,11 @@ spec:
   template:
     metadata:
       annotations:
-        {{- if empty .Values.global.cliVersion }}
+        {{ if kindIs "string" .Values.stage -}}
+        {{ if not (eq .Values.stage "control-plane") -}}
```

Comment:
Given `.Values.stage` won't be present for Helm installs, `checksum/config` won't be produced in that case and will break the mechanism for rolling-out the pod :-(

```bash
$ helm install linkerd -f ~/k8s/helm/linkerd-overrides.yaml charts/linkerd2
$ helm upgrade linkerd -f ~/k8s/helm/linkerd-overrides.yaml --set global.enableEndpointSlices=true charts/linkerd2
```
The last command will produce a restart only for the destination pod.
And these are the logs for the injector:
```bash
$ k -n linkerd logs -f linkerd-proxy-injector-7896f8d647-524nc proxy-injector
time="2020-12-09T22:02:34Z" level=info msg="running version dev-f7e2c84d-alpeb"
time="2020-12-09T22:02:34Z" level=info msg="waiting for caches to sync"
time="2020-12-09T22:02:35Z" level=info msg="caches synced"
time="2020-12-09T22:02:35Z" level=info msg="listening at :8443"
time="2020-12-09T22:02:35Z" level=info msg="starting admin server on :9995"
2020/12/09 22:04:41 http: TLS handshake error from 127.0.0.1:47788: remote error: tls: bad certificate
time="2020-12-09T22:04:47Z" level=info msg="Updated certificate"
```
It ends up working at the end only because of the dynamic cert reloading.

Reply:
Bringing back `{{- if empty .Values.global.cliVersion }}` in addition to the current conditions could do the trick, for always producing the checksum for Helm installs.

Reply:
Okay. That seems to be a problem. How about we set `stage:""` in `values.yaml` so that this is added even in helms way.
Seems to work.

```bash
tarun in dev in on  kind-kind () linkerd2 on  tarun/webhooks-upgrade [$!?] via 🐹 v1.15.4
❯ ./bin/helm upgrade linkerd2-1607576483   charts/linkerd2 \
  --set-file global.identityTrustAnchorsPEM=ca.crt \
  --set-file identity.issuer.tls.crtPEM=issuer.crt \
  --set-file identity.issuer.tls.keyPEM=issuer.key \
  --set global.enableEndpointSlices=true \
  --set identity.issuer.crtExpiry=$exp
Release "linkerd2-1607576483" has been upgraded. Happy Helming!
NAME: linkerd2-1607576483
LAST DEPLOYED: Thu Dec 10 05:04:05 2020
NAMESPACE: default
STATUS: deployed
REVISION: 2
TEST SUITE: None
NOTES:
The Linkerd control plane was successfully installed 🎉

To help you manage your Linkerd service mesh you can install the Linkerd CLI by running:

  curl -sL https://run.linkerd.io/install | sh

Alternatively, you can download the CLI directly via the Linkerd releases page:

  https://github.com/linkerd/linkerd2/releases/

Looking for more? Visit https://linkerd.io/2/getting-started/

tarun in dev in on  kind-kind () linkerd2 on  tarun/webhooks-upgrade [$!?] via 🐹 v1.15.4 took 6s
❯

tarun in dev in on  kind-kind () linkerd2 on  tarun/webhooks-upgrade [$!?] via 🐹 v1.15.4 took 6s
❯ kgp -A
NAMESPACE            NAME                                         READY   STATUS            RESTARTS   AGE
kube-system          coredns-f9fd979d6-g7j6t                      1/1     Running           0          7m42s
kube-system          coredns-f9fd979d6-tkv76                      1/1     Running           0          7m42s
kube-system          etcd-kind-control-plane                      1/1     Running           0          7m52s
kube-system          kindnet-pwz9z                                1/1     Running           0          7m42s
kube-system          kube-apiserver-kind-control-plane            1/1     Running           0          7m52s
kube-system          kube-controller-manager-kind-control-plane   1/1     Running           0          7m52s
kube-system          kube-proxy-vl5mk                             1/1     Running           0          7m42s
kube-system          kube-scheduler-kind-control-plane            1/1     Running           0          7m52s
linkerd              linkerd-controller-99d7c455f-6w2dr           2/2     Running           0          2m41s
linkerd              linkerd-destination-545db4cdbb-ndz8b         0/2     PodInitializing   0          3s
linkerd              linkerd-destination-f669c68d9-zlnbq          2/2     Running           0          2m40s
linkerd              linkerd-grafana-59f85f5795-lnrsm             2/2     Running           0          2m41s
linkerd              linkerd-identity-7d5f6d797b-swcbd            2/2     Running           0          2m41s
linkerd              linkerd-prometheus-5669f85bf-xnj57           2/2     Running           0          2m40s
linkerd              linkerd-proxy-injector-5db757445b-rcvcz      2/2     Running           0          2m41s
linkerd              linkerd-proxy-injector-d85d5b5f7-vlfct       0/2     PodInitializing   0          3s
linkerd              linkerd-sp-validator-79c54c76c5-2jj6w        0/2     Init:0/1          0          3s
linkerd              linkerd-sp-validator-8547f97d45-dsbhf        2/2     Running           0          2m41s
linkerd              linkerd-tap-59bd44bc6c-fw7w4                 2/2     Running           0          2m40s
linkerd              linkerd-tap-78956c47dc-hjlcl                 0/2     Init:0/1          0          2s
linkerd              linkerd-web-65cd6786b8-2qbrg                 2/2     Running           0          2m41s
local-path-storage   local-path-provisioner-78776bfc44-jhn48      1/1     Running           0          7m42s
```